### ISSUE TITLE
bind is always listed in the import suggestions for Prelude

### ISSUE THREAD

**Issue by garyb** [purescript/purescript#1739]

**Comment by paf31** [comment]
Because of `do` desugaring surely?

**Comment by garyb** [comment]
No, it's listed even if there is no `do` usage. It's down to the special case handling of it in the unused name detection I think.

**Comment by nwolverson** [comment]
I think the unused name detection basically has to be pushed later in the pipeline but the existing info (the originally qualified usage) retained. (Or some 'look for `do`' hack...)

**Comment by garyb** [comment]
Yeah, I was thinking of looking for a `do` :)

**Comment by paf31** [comment]
Can someone explain why `bind` is special again please?

**Comment by garyb** [comment]
Actually, I don't know. Things seem to work as they should with the special case removed... @nwolverson?

**Comment by nwolverson** [comment]
Oh cool, I'm happy to be proved wrong. I was going to say it's because `bind` desugaring happens later in the desugaring pipeline but I just looked. Repeatedly. And it happens earlier.

Not sure why I thought this was the case but I think I convinced myself this was required because I tried a test case too simple:

```
do
    log "Hello world"
```

Which of course actually doesn't use `bind` at all. 

So *yes*, I think things work as they are without the `sugarNames` special case, and this also works for `bind` imported from somewhere other than `Prelude`.
